benchmarks/CMakeLists.txt

```diff
@@ -37,6 +37,16 @@ link_directories("${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}" # CMAKE_CUDA_IMPLICIT
                  "${GTEST_LIBRARY_DIR}"
                  "${GBENCH_LIBRARY_DIR}")
 
+###################################################################################################
+# - build options ---------------------------------------------------------------------------------
+
+option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
```

Comment:
I think the tests and benchmarks should inherit the setting from the main cmakelists.txt.  It doesn't make sense to build the tests with it if it is disabled for the library.

Reply:
Done.

Reply:
Looks like you removed it completely from the benchmarks and tests rather than inheriting it? Does it get set correctly when compiling a test binary, for example?

Reply:
`add_subdirectory` inherits the options from the parent file, but only at the point of calling it. That's why I moved the PTDS option up, before adding tests/benchmarks. See https://stackoverflow.com/a/7980137.